mac/ZRTOPTBL.CPY

```diff
@@ -0,0 +1,316 @@
+         B     GETHLASM_READ           * go process the entries
+
+GETHLASM_PROCESS DS 0H
+*
+* Ignore any lines that have a non-blank character in position 2
+* (column 1 is reserved for the ASA printer control character)
+         CLI   HLASMREC+1,C' '         * blank char in second pos?
+         BE    GETHLASM_READ           * yes: get next record
+*
+* Ignore any lines having the marker
+         CLC   MARKER,HLASMREC+1       * Record has marker?
+         BE    GETHLASM_READ           * Yes: ignore record
+*
+* Should be a valid line with up to three entries
+         LA    R4,HLASMREC+1           * point first entry
+GETHLASM_PROCENT DS 0H                 * Process an entry
+         CLC   LASTOPCD,0(R4)          * Compare mnemonics
+         BNL   GETHLASM_END            * End of list reached!
+         C     R5,=AL4(HLASMTBL_END)   * Table overflow?
+         BNL   ERROR05                 * Oh no - need bigger table!
+         MVC   0(L'MARKER,R5),0(R4)    * Copy the entry
+         MVC   LASTOPCD,0(R4)          * update last mnemonic
+         LA    R5,L'MARKER(,R5)        * Bump buffer pointer
+         LA    R4,L'MARKER(,R4)        * Bump input pointer
+         C     R4,=AL4(HLASMREC+1+3*L'MARKER) * record done?
+         BL    GETHLASM_PROCENT        * no: process next entry
+         B     GETHLASM_READ           * yes: get next record
+
+GETHLASM_END   DS 0H
+         ST    R5,HLASMMPT             * Save ptr to empty entry
+         MVC   0(8,R5),=CL8'99999999'  * terminator mnemonic
+         L     R14,GETHLASM_RET        * Get return address
+         BR    R14                     * Return
+
+***********************************************************************
+*
+* Compare routine for extracted opcode definitions
+*
+***********************************************************************
+COMPARE  DS    0H
+         ST    R14,COMPARE_RET          * Save return address
+
+         L     R4,=AL4(Z390TBL)         * Point entries buffer
+         L     R5,Z390MPT               * end of buffer
+         L     R6,=AL4(HLASMTBL)        * Point entries buffer
+         L     R7,HLASMMPT              * Table overflow?
+
+COMPARE_LOOP   DS 0H
+         CLC   0(8,R4),0(R6)            * Compare mnemonics
+         BL    ERROR06                  * Superfluous z390 opcode
+         BH    ERROR07                  * Missing z390 opcode
+         CLC   0(&ENTRYLEN.,R4),0(R6)   * Compare entire entry
+         BNE   ERROR08                  * Report incorrec definition
+         LA    R4,&ENTRYLEN.(,R4)       * Bump z390 entry pointer
+         LA    R6,&ENTRYLEN.(,R6)       * Bump hlasm entry pointer
+COMPARE_NEXT   DS 0H                    * advance to next entry
+         CLR   R4,R5                    * z390 table exhausted?
+         BL    COMPARE_LOOP             * no: compare next entry
+         CLR   R6,R7                    * hlasm table exhausted?
+         BL    COMPARE_LOOP             * no: compare next entry
```

Comment:
Observation.

If the z390 table is larger than the HLASM table, then the first "BL" might be using an invalid R6???
If the HLASM table is larger than the z390 table, then the second "BL" might be using an invalid R4???

Your artificial "999999999" entries at the ends of the tables may prevent these possibilities.